x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+      };
+
+      it('should replace the existing token with the new token by name', () => {
+        mount({
+          apiTokens: [newToken, existingToken],
+        });
+        const updatedExistingToken = {
+          ...existingToken,
+          type: ADMIN,
+        };
+
+        CredentialsLogic.actions.onApiTokenUpdateSuccess(updatedExistingToken);
+        expect(CredentialsLogic.values.apiTokens).toEqual([newToken, updatedExistingToken]);
+      });
+
+      // TODO Not sure if this is a good behavior or not
+      it('if for some reason the existing token is not found, it adds a new token...', () => {
+        mount({
+          apiTokens: [newToken, existingToken],
+        });
+        const brandNewToken = {
+          name: 'brand new token',
+          type: ADMIN,
+        };
+
+        CredentialsLogic.actions.onApiTokenUpdateSuccess(brandNewToken);
+        expect(CredentialsLogic.values.apiTokens).toEqual([newToken, existingToken, brandNewToken]);
+      });
+    });
+
+    describe('activeApiToken', () => {
+      it('should reset to the default value, which effectively clears out the current form', () => {
+        mount({
+          activeApiToken: newToken,
+        });
+
+        CredentialsLogic.actions.onApiTokenUpdateSuccess({ ...newToken, type: ADMIN });
+        expect(CredentialsLogic.values.activeApiToken).toEqual(DEFAULT_VALUES.activeApiToken);
+      });
+    });
+
+    describe('activeApiTokenRawName', () => {
+      it('should reset to the default value, which effectively clears out the current form', () => {
+        mount({
+          activeApiTokenRawName: 'foo',
+        });
+
+        CredentialsLogic.actions.onApiTokenUpdateSuccess({ ...newToken, type: ADMIN });
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual(
+          DEFAULT_VALUES.activeApiTokenRawName
+        );
+      });
+    });
+
+    describe('showCredentialsForm', () => {
+      it('should reset to the default value, which closes the credentials form', () => {
+        mount({
+          showCredentialsForm: true,
+        });
+
+        CredentialsLogic.actions.onApiTokenUpdateSuccess({ ...newToken, type: ADMIN });
+        expect(CredentialsLogic.values.showCredentialsForm).toEqual(false);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.onApiTokenUpdateSuccess({ ...newToken, type: ADMIN });
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        apiTokens: expect.any(Array),
+        activeApiToken: expect.any(Object),
+        activeApiTokenRawName: expect.any(String),
+        showCredentialsForm: expect.any(Boolean),
+      });
+    });
+  });
+
+  describe('setCredentialsData', () => {
+    const meta: IMeta = {
+      page: {
+        current: 1,
+        size: 1,
+        total_pages: 1,
+        total_results: 1,
+      },
+    };
+
+    describe('apiTokens', () => {
+      it('should set', () => {
```

Comment:
Should this should `be` set to match the meta test below?

```suggestion
      it('should be set', () => {
```